**Aplicaciones en ciencia de datos**

**Laboratorio 3**

María Paula Angulo Collantes - 202212044

Danna Gabriela Castro Villalba - 202615188

In [16]:
import pandas as pd

**Punto 1**

In [17]:
#Carga
df = pd.read_csv('estudiantes - Copy.csv')

In [ ]:
#Primeras filas
df.head()

,student_id,programa,semestre,edad,genero,horas_estudio_semana,horas_sueno_noche,tiempo_desplazamiento_min,medio_transporte,nivel_estres,satisfaccion_academica,dias_ejercicio_semana,horas_sociales_semana,uso_ia_generativa,confianza_programando,area_interes_datos,modalidad_preferida,promedio_acumulado,fecha_encuesta
0,E0001,Economía,5,21,Mujer,16.8,5.5,61.0,Bus,4,4,3,7.2,A veces,3,Ingeniería de datos,Presencial,3.47,2026-08-01
1,E0002,Administración,6,23,Mujer,23.5,7.2,98.0,TransMilenio,4,4,2,10.8,Nunca,2,No estoy seguro,Presencial,3.69,2026-08-10
2,E0003,Administración,1,17,Hombre,12.6,6.7,55.0,TransMilenio,5,4,3,6.8,A veces,4,Analítica de negocio,Presencial,3.26,2026-08-21
3,E0004,Ciencia de Datos,8,21,Mujer,15.7,6.1,57.0,Moto,4,3,1,14.3,Nunca,4,Analítica de negocio,Presencial,2.90,2026-08-19
4,E0005,Diseño,3,22,Mujer,20.5,6.4,50.0,Carro,5,4,5,11.0,A veces,2,No estoy seguro,Presencial,3.54,2026-08-04


In [8]:
#Últimas filas
df.tail()

,student_id,programa,semestre,edad,genero,horas_estudio_semana,horas_sueno_noche,tiempo_desplazamiento_min,medio_transporte,nivel_estres,satisfaccion_academica,dias_ejercicio_semana,horas_sociales_semana,uso_ia_generativa,confianza_programando,area_interes_datos,modalidad_preferida,promedio_acumulado,fecha_encuesta
2507,E2113,Ciencia de Datos,1,18,Mujer,13.7,6.3,81.0,Caminando,5,4,5,9.7,Frecuentemente,3,Analítica de negocio,Presencial,3.67,2026-08-10
2508,E0456,Administración,2,19,Mujer,17.7,5.8,47.0,Caminando,4,4,3,4.9,Frecuentemente,1,IA y ML,Presencial,3.24,2026-08-04
2509,E1001,Ingeniería de Sistemas,1,19,Mujer,20.1,5.2,92.0,Caminando,5,3,1,9.1,Frecuentemente,4,Analítica de negocio,Presencial,3.82,2026-08-19
2510,E0438,Ingeniería Industrial,1,19,Mujer,15.5,7.8,50.0,TransMilenio,3,4,5,3.7,Todos los días,3,Analítica de negocio,Presencial,3.79,2026-08-11
2511,E1192,Administración,2,20,Mujer,15.6,7.7,50.0,Caminando,3,4,6,10.8,A veces,4,Visualización,Presencial,3.75,2026-08-02


In [10]:
#Dimensiones del DataFrame
df.shape

(2512, 19)

In [11]:
#Nombre de las columnas
df.columns

Index(['student_id', 'programa', 'semestre', 'edad', 'genero',
       'horas_estudio_semana', 'horas_sueno_noche',
       'tiempo_desplazamiento_min', 'medio_transporte', 'nivel_estres',
       'satisfaccion_academica', 'dias_ejercicio_semana',
       'horas_sociales_semana', 'uso_ia_generativa', 'confianza_programando',
       'area_interes_datos', 'modalidad_preferida', 'promedio_acumulado',
       'fecha_encuesta'],
      dtype='str')

In [12]:
#Tipos de datos inferidos por Pandas
df.dtypes

student_id                       str
programa                         str
semestre                       int64
edad                           int64
genero                           str
horas_estudio_semana         float64
horas_sueno_noche            float64
tiempo_desplazamiento_min    float64
medio_transporte                 str
nivel_estres                   int64
satisfaccion_academica         int64
dias_ejercicio_semana          int64
horas_sociales_semana        float64
uso_ia_generativa                str
confianza_programando          int64
area_interes_datos               str
modalidad_preferida              str
promedio_acumulado           float64
fecha_encuesta                   str
dtype: object

**Preguntas**

a. ¿Qué representa una fila del dataset?

**Rta:** Cada fila del dataset representa a un estudiante individual. Cada registro está identificado por un código único y describe detalladamente las respuestas de un único alumno a la encuesta.

b. ¿Cuál es la unidad de análisis?

**Rta:** La unidad de análisis es el estudiante.

c. ¿Cuántas observaciones y variables hay?

**Rta:** Hay 2512 observaciones (filas), cada una de ellas corresponde a un estudiante único encuestado y 19 variables (columnas), cada una representa una característica medida o registrada de los estudiantes.

d. ¿Qué parece representar el dataset en términos del problema o dominio?

**Rta:** En términos del problema y su dominio, este dataset representa un diagnóstico integral y multidimensional de la vida y el bienestar de la comunidad estudiantil en un entorno de educación superior.

**Punto 2**

In [23]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2512 entries, 0 to 2511
Data columns (total 19 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   student_id                 2512 non-null   str    
 1   programa                   2512 non-null   str    
 2   semestre                   2512 non-null   int64  
 3   edad                       2512 non-null   int64  
 4   genero                     2512 non-null   str    
 5   horas_estudio_semana       2512 non-null   float64
 6   horas_sueno_noche          2440 non-null   float64
 7   tiempo_desplazamiento_min  2456 non-null   float64
 8   medio_transporte           2512 non-null   str    
 9   nivel_estres               2512 non-null   int64  
 10  satisfaccion_academica     2512 non-null   int64  
 11  dias_ejercicio_semana      2512 non-null   int64  
 12  horas_sociales_semana      2512 non-null   float64
 13  uso_ia_generativa          2512 non-null   str    
 14  con

In [25]:
df.nunique()

student_id                   2500
programa                        8
semestre                       10
edad                           13
genero                          5
horas_estudio_semana          305
horas_sueno_noche              57
tiempo_desplazamiento_min     135
medio_transporte                7
nivel_estres                    5
satisfaccion_academica          4
dias_ejercicio_semana           8
horas_sociales_semana         189
uso_ia_generativa               4
confianza_programando           5
area_interes_datos              6
modalidad_preferida             3
promedio_acumulado            191
fecha_encuesta                 21
dtype: int64

| Variables                 | Tipo en Python | Escala   | Descripción                                             | Ejemplo             | Observaciones                                                                                                     |
|:-------------------------:|:--------------:|:--------:|:-------------------------------------------------------:|:-------------------:|:-----------------------------------------------------------------------------------------------------------------:|
| student_id                | str            | Nominal  | Identificador único del estudiante                      | E0001               | No es único                                                                                                       |
| programa                  | str            | Nominal  | Programa académico del estudiante                       | Economía            |                                                                                                                   |
| semestre                  | int64          | Ordinal  | Semestre que cursa el estudiante                        | 5                   | Va de 1 - 10                                                                                                      |
| edad                      | int64          | Discreta | Edad del estudiante en años                             | 21                  | Va de 16 - 28                                                                                                     |
| genero                    | str            | Nominal  | Genero con el que se identifica el estudiante           | Mujer               |                                                                                                                   |
| horas_estudio_semana      | float64        | Continua | Horas que dedica a estudiar                             | 16.8                | Rango observado: 0 a 95 horas; el valor máximo (95) es alto — verificar si es un dato real o un error de captura. |
| horas_sueno_noche         | float64        | Continua | Horas que duerme por la noche                           | 5.5                 | 72 valores nulos                                                                                                  |
| tiempo_desplazamiento_min | float64        | Continua | Tiempo de desplazamiento al lugar de estudio en minutos | 61.0                | 56 valores nulos. Valor máximo de 360 min (6 horas) posible outlier o error de captura                            |
| medio_transporte          | str            | Nominal  | Medio de transporte usado                               | Bus                 |                                                                                                                   |
| nivel_estres              | int64          | Ordinal  | Nivel de estrés autopercibido                           | 4                   | Va de 1 - 5                                                                                                       |
| satisfaccion_academica    | int64          | Ordinal  | Nivel de satisfacción con la vida académica             | 4                   | Va de 1 - 5                                                                                                       |
| dias_ejercicio_semana     | int64          | Discreta | Número de días a la semana que hace ejercicio           | 3                   | Va de 0 - 7                                                                                                       |
| horas_sociales_semana     | float64        | Continua | Número de horas a la semana que dedica a socializar     | 7.2                 | Va de 0 - 22.7                                                                                                    |
| uso_ia_generativa         | str            | Ordinal  | Frecuencia de herramientas de IA                        | A veces             | Categorías: Nunca < A veces < Frecuentemente < Todos los días                                                     |
| confianza_programando     | int64          | Ordinal  | Nivel de confianza autopercibido en programación        | 3                   | Va de 1 - 5                                                                                                       |
| area_interes_datos        | str            | Nominal  | Área de interés dentro de ciencia de datos              | Ingeniería de datos | 41 Valores nulos                                                                                                  |
| modalidad_preferida       | str            | Nominal  | Modalidad de estudio preferida                          | Presencial          | Categorías: Presencial, Híbrida, Virtual.                                                                         |
| promedio_acumulado        | float64        | Continua | Promedio académico acumulado del estudiante             | 3.47                | Va de 2.36 - 4.73                                                                                                 |
| fecha_encuesta            | str            | -        | Fecha en la que se respondio la encuenta                | 2026-08-01          | Está almacenada como texto (str), no como tipo datetime                                                           |


**Punto 3**

In [41]:
#Dimensiones 
print("Dimensiones del DataFrame:", df.shape)
#Tipos
print("Tipos de datos de las columnas:\n",
      
      df.dtypes,

      "\nSin embargo,fecha_encuesta se infirió como str en lugar de un tipo de fecha")
#Faltantes
print("Los valores faltantes se encuentran en las columnas: horas_sueno_noche, tiempo_desplazamiento, area_interes_datos")
#Duplicados
print("Hay",df.duplicated().sum(),"filas duplicadas en el DataFrame")
#Cardinalidad
print("Cardinalidad de las columnas:\n",df.nunique())
#Distribución
print("Distribución de las columnas:\n", df.describe())

Dimensiones del DataFrame: (2512, 19)
Tipos de datos de las columnas:
 student_id                       str
programa                         str
semestre                       int64
edad                           int64
genero                           str
horas_estudio_semana         float64
horas_sueno_noche            float64
tiempo_desplazamiento_min    float64
medio_transporte                 str
nivel_estres                   int64
satisfaccion_academica         int64
dias_ejercicio_semana          int64
horas_sociales_semana        float64
uso_ia_generativa                str
confianza_programando          int64
area_interes_datos               str
modalidad_preferida              str
promedio_acumulado           float64
fecha_encuesta                   str
dtype: object 
Sin embargo,fecha_encuesta se infirió como str en lugar de un tipo de fecha
Los valores faltantes se encuentran en las columnas: horas_sueno_noche, tiempo_desplazamiento, area_interes_datos
Hay 12 filas duplicad

**Punto 4:** ¿Hay columnas cuyo significado o nombre no sea suficientemente claro?

Sí, se encontraron varias. Por ejemplo, en nivel_estres, satisfaccion_academica y confianza_programando el nombre no especifica qué significa cada extremo de la escala (no queda claro si 1 representa "bajo" y 5 "alto", o al revés). En tiempo_desplazamiento_min no se aclara si el desplazamiento es hacia la universidad o hacia otro lugar.

**Punto 5**

In [16]:
df["genero"].value_counts(False)

genero
Hombre                   1238
Mujer                    1148
Prefiere no responder      72
No binario                 47
mujer                       7
Name: count, dtype: int64

In [19]:
df["genero"].value_counts(dropna=False)

genero
Hombre                   1238
Mujer                    1148
Prefiere no responder      72
No binario                 47
mujer                       7
Name: count, dtype: int64

In [20]:
df["medio_transporte"].value_counts(dropna=False)

medio_transporte
TransMilenio    788
Bus             501
Carro           442
Caminando       330
Bicicleta       252
Moto            188
Transmilenio     11
Name: count, dtype: int64

In [21]:
df.isna().sum()

student_id                    0
programa                      0
semestre                      0
edad                          0
genero                        0
horas_estudio_semana          0
horas_sueno_noche            72
tiempo_desplazamiento_min    56
medio_transporte              0
nivel_estres                  0
satisfaccion_academica        0
dias_ejercicio_semana         0
horas_sociales_semana         0
uso_ia_generativa             0
confianza_programando         0
area_interes_datos           41
modalidad_preferida           0
promedio_acumulado            0
fecha_encuesta                0
dtype: int64

**Punto 5: (Paula)** ¿Existen códigos, categorías o unidades que requerirían documentación adicional?

Sí. El campo student_id usa un código tipo "E0001" cuyo formato o lógica de asignación no está documentado (no se sabe si el número corresponde a un orden de inscripción, un consecutivo aleatorio, u otro criterio). Las escalas numéricas de nivel_estres, satisfaccion_academica y confianza_programando tampoco cuentan con documentación sobre qué representa cada valor del 1 al 5, lo cual dificulta interpretar correctamente los resultados. La unidad de tiempo_desplazamiento_min se puede inferir por el sufijo "_min" (minutos), pero no se especifica si corresponde a un trayecto de ida, de ida y vuelta, o si varía según el día de la semana. La categoría "No estoy seguro" en area_interes_datos también requeriría documentación adicional para precisar si debe tratarse como una respuesta válida o como un valor faltante. Por último, no hay documentación sobre la metodología o el rango completo de promedio_acumulado (por ejemplo, si la escala oficial va de 0.0 a 5.0, y qué unidad académica lo calcula y reporta).